torchrec/csrc/dynamic_embedding/details/ctz_impl.cpp

```diff
@@ -0,0 +1,86 @@
+#include <torchrec/csrc/dynamic_embedding/details/bits_op.h>
+
+namespace torchrec::dynamic_embedding::details::bits_impl {
+
+template <typename T>
+struct CtzImpl {
+  /**
+   * Naive implementation for no __builtin_ctz
+   * @param v
+   * @return
+   */
+  int operator()(T v) const {
+    if (v == 0)
+      return -1;
+    int result = 0;
+    while ((v & 1) == 0) {
+      v >>= 1;
+      result += 1;
+    }
+    return result;
+  }
+};
+
+#if defined(__GNUC__) || defined(__clang__)
```

Comment:
I don't think we support any other compilers so it's fine not to even have ifdef here

Reply:
removed.